src/rgw/rgw_kms.cc

```diff
@@ -0,0 +1,275 @@
+// -*- mode:C; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+/**
+ * Server-side encryption integrations with Key Management Systems (SSE-KMS)
+ */
+
+#include <rgw/rgw_crypt.h>
+#include <rgw/rgw_keystone.h>
+#include <rgw/rgw_b64.h>
+#include "include/str_map.h"
+#include "common/safe_io.h"
+#include "rgw/rgw_kms.h"
+
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_rgw
+
+using namespace rgw;
+
+
+map<string,string> get_str_map(const string &str) {
+  map<string,string> m;
+  get_str_map(str, &m, ";, \t");
+  return m;
+}
+
+int get_actual_key_from_conf(CephContext *cct,
+                             boost::string_view key_id,
+                             boost::string_view key_selector,
+                             std::string& actual_key)
+{
+  int res = 0;
+
+  static map<string,string> str_map = get_str_map(
+      cct->_conf->rgw_crypt_s3_kms_encryption_keys);
+
+  map<string, string>::iterator it = str_map.find(std::string(key_id));
+  if (it == str_map.end())
+    return -ERR_INVALID_ACCESS_KEY;
+
+  std::string master_key;
+  try {
+    master_key = from_base64((*it).second);
+  } catch (...) {
```

Comment:
please catch as (std::exception&)

Reply:
This is part of the existing code we moved from rgw_crypto (to keep all SSE-KMS functions in one file). I tried not to touch that code too much as there are no unit tests and I'm not too sure how to test it manually, but your suggestion looks safe enough to change.